### DISCUSSION TITLE
Is it possible to request a file type be added to GitHub?

### DISCUSSION THREAD

**Discussion by Enderbyte09** [community/community#84984] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have written a custom file type and I was wondering, is there a way to get the custom file type indexed by Github so that it appears on that code makeup bar (That says something like: Python 99% Shell 1% or the like)?

**Comment by nickkipshidze** [comment]
The [Linguist library](https://github.com/github-linguist/linguist) does all the calculations on your GitHub repo to show you the code percentages and all that. 

You can change the language of a repository in `.gitattributes`, see https://stackoverflow.com/questions/13597892/how-can-i-change-the-language-of-a-repository-on-github/13599737#13599737. But if your custom language is not supported by Linguist, you can contribute to the Lingust project https://help.github.com/articles/about-repository-languages/. 

Here are the languages supported by Linguist: https://github.com/github/linguist/blob/master/vendor/README.md

I hope that helps.
